RFC-0006-nan-definition.md

```diff
@@ -0,0 +1,179 @@
+
+The disadvantage is that we aren't strictly following the IEEE-754 definitions.
+
+We propose changing the behavior of all of our functinos and operators to match this definition (see the spreadsheet linked in the description of Presto's current behavior for the full list). Notably, this will involve changing basic operators (<, >, =, <>) and joins.  another important thing to call out is that Presto may read files and metadata that were written by other systems and it can write files and metadata that will be read by other systems. It is important to ensure that, for example, when Presto writes stats for max values it follows the expectations for the service it is writing too, and when it reads max values, it knows whether those will adhere to its own NaN ordering definition before using that metadata for its own optimization.
+
+We will also add thorough documentation for the new behavior.
+ 
+## Other Approaches Considered
+
+### IEEE-754 and throw an error for order by
+
+In this approach we follow the strict IEEE-754 definition for equality and comparisons. Meaning NaN = NaN would be false.  NaN less than or greater than anything is false and NaN <> NaN is true.  Any function on NaN input including min/max will return NaN. Grouping or distinct on NaN keys will return separate groups for each nan value. In cases where we could not simply return NaN, we would throw an exception.  That would mean that for order by on a double column containing NaN we would throw an error (as presumably would array_sort). 
+
+This has the advantage of adhering to IEEE. It also provides a consistent definition of our operators to reduce chances of bugs (no difference between results you would get from applying < or > vs. what you would get from doing a sort).
+
+A disadvantage is that users are very unlikely to want to get an error for order by just because their data has NaN in it- especially if we otherwise support NaN values. it also still leaves the funny behavior with maps/hashing where you can put things in a map but can't get it out. There are also no other systems that behave this way.
```

Comment:
```suggestion
A disadvantage is that users are very unlikely to want to get an error for order by just because their data has NaN in it - especially if we otherwise support NaN values. it also still leaves the funny behavior with maps/hashing where you can put things in a map but can't get it out. There are also no other systems that behave this way.
```